cli/cmd/proxy/read/filters_test.go

```diff
@@ -0,0 +1,508 @@
+					Type:                     "STATIC",
+					LastUpdated:              "2022-05-13T04:22:39.553Z",
+				},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			actual := FilterClusters(given, tc.fqdn, tc.address, tc.port)
+			require.Equal(t, tc.expected, actual)
+		})
+	}
+}
+
+func TestFilterEndpoints(t *testing.T) {
+	given := []Endpoint{
+		{
+			Address: "192.168.79.187:8502",
+			Cluster: "local_agent",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "127.0.0.1:8080",
+			Cluster: "local_app",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.31.201:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.47.235:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+		{
+			Address: "192.168.71.254:20000",
+			Weight:  1,
+			Status:  "HEALTHY",
+		},
+	}
+
+	cases := map[string]struct {
+		address  string
+		port     int
+		expected []Endpoint
+	}{
+		"No filter": {
+			address: "",
+			port:    -1,
+			expected: []Endpoint{
+				{
+					Address: "192.168.79.187:8502",
+					Cluster: "local_agent",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+				{
+					Address: "127.0.0.1:8080",
+					Cluster: "local_app",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+				{
+					Address: "192.168.31.201:20000",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+				{
+					Address: "192.168.47.235:20000",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+				{
+					Address: "192.168.71.254:20000",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+			},
+		},
+		"Filter address": {
+			address: "127.0.0.1",
+			port:    -1,
+			expected: []Endpoint{
+				{
+					Address: "127.0.0.1:8080",
+					Cluster: "local_app",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+			},
+		},
+		"Filter port": {
+			address: "",
+			port:    20000,
+			expected: []Endpoint{
+				{
+					Address: "192.168.31.201:20000",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+				{
+					Address: "192.168.47.235:20000",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+				{
+					Address: "192.168.71.254:20000",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+			},
+		},
+		"Filter address and port": {
+			address: "235",
+			port:    20000,
+			expected: []Endpoint{
+				{
+					Address: "192.168.47.235:20000",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			actual := FilterEndpoints(given, tc.address, tc.port)
+			require.Equal(t, tc.expected, actual)
+		})
+	}
+}
+
+func TestFilterListeners(t *testing.T) {
+	given := []Listener{
+		{
+			Name:    "public_listener",
+			Address: "192.168.69.179:20000",
+			FilterChain: []FilterChain{
+				{
+					FilterChainMatch: "Any",
+					Filters:          []string{"* -> local_app/"},
+				},
+			},
+			Direction:   "INBOUND",
+			LastUpdated: "2022-06-09T00:39:27.668Z",
```

Comment:
are these important for testing? Might be worth omitting them to improve readability if not.

Reply:
That is a great call.